Framework/DataHandling/src/SaveDiffFittingAscii.cpp

```diff
@@ -0,0 +1,246 @@
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidDataHandling/SaveDiffFittingAscii.h"
+
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/TableRow.h"
+#include "MantidDataObjects/TableWorkspace.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "Poco/File.h"
+#include <boost/tokenizer.hpp>
+#include <fstream>
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace Kernel;
+using namespace Mantid::API;
+
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(SaveDiffFittingAscii)
+
+using namespace Kernel;
+using namespace API;
+
+/// Empty constructor
+SaveDiffFittingAscii::SaveDiffFittingAscii()
+    : Mantid::API::Algorithm(), m_sep(','), m_endl('\n'), m_counter(0) {}
+
+/// Initialisation method.
+void SaveDiffFittingAscii::init() {
+
+  declareProperty(make_unique<WorkspaceProperty<ITableWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "The name of the workspace containing the data you want to "
+                  "save to a TBL file");
+
+  // Declare required parameters, filename with ext {.his} and input
+  // workspace
+  const std::vector<std::string> exts{".txt", ".csv", ""};
+  declareProperty(Kernel::make_unique<API::FileProperty>(
+                      "Filename", "", API::FileProperty::Save, exts),
+                  "The filename to use for the saved data");
+
+  declareProperty(
+      "RunNumber", "", boost::make_shared<MandatoryValidator<std::string>>(),
+      "Run number list of the focused files, which is used to generate the "
+      "parameters table workspace");
+
+  declareProperty(
+      "Bank", "", boost::make_shared<MandatoryValidator<std::string>>(),
+      "Bank number list of the focused files, which is used to generate "
+      "the parameters table workspace");
+
+  std::vector<std::string> formats;
+
+  formats.push_back("AppendToExistingFile");
+  formats.push_back("WriteGroupWorkspace");
+  formats.push_back("OverwriteFile");
+  declareProperty(
+      "OutFormat", "AppendToExistingFile",
```

Comment:
Very good set of alternatives. Is this `OutFormat` or `OutMode` or similar? `OutFormat` could be confusing.